src/interpreter/vm/VM.cpp

```diff
@@ -128,14 +151,36 @@ int VM::run() {
                     std::cout << "TypeError: Invalid operand type for -: "
                               << a.type
                               << "." << std::endl;
+                    exit(EXIT_FAILURE);
+                }
+                break;
+            case Opcode::NOT:
+                a = pop();
+                if (a.type == DurianType::Boolean) {
+                    push(!a.value.bval);
+                } else {
+                    std::cout << "TypeError: Invalid operand type for !: "
+                              << a.type
+                              << "." << std::endl;
+                    exit(EXIT_FAILURE);
+                }
+                break;
+            case Opcode::BR_F:
+                a = pop();
+                jumpLen = *reinterpret_cast<int32_t*>(m_code+m_pc);
+                m_pc += sizeof(int32_t);
+                if (a.isFalsy()) {
+                    m_pc += jumpLen;
                 }
                 break;
             case Opcode::PRINT:
                 a = pop();
                 if (a.type == DurianType::Integer)
                     std::cout << a.value.ival << std::endl;
                 else if (a.type == DurianType::Double)
-                    std::cout << a.value.dval << std::endl;
+                    printf("%f\n", a.value.dval);
```

Comment:
are we using both `std::cout` and `printf`?

Reply:
I couldn't remember how to print floating point off the top of my head when @CoffeeTableEspresso asked, so he just did `printf` instead. Personally I don't mind having both since `printf` can be shorter and they're both "obviously" printing.